cli: add an environment variable to disable telemetry globally

folliehiyuki opened the pull request:
With the current model described at <https://devpod.sh/docs/other-topics/telemetry#telemetry-opt-out-process>, a CLI user wanting to opt out of telemetry completely needs to remember to set the TELEMTRY option for every new context.

This PR adds an additional environment variable as a method to disable telemetry regardless of the current context. This is a common approach among open-sourced CLI programs.

pascalbreuninger commented:
@folliehiyuki thanks for your contribution.
Fair enough, could you also update the docs please?

folliehiyuki commented:
I've updated the telemetry docs page. My editor spotted 2 spelling errors in the file, so I added another commit fixing those as well, if you don't mind.

pascalbreuninger reviewed (approved):
LGTM
